test/xbgpu/test_engine.py

```diff
@@ -695,8 +668,15 @@ async def _send_data(
 
                 corrprod_results[i][j] = ig_recv["xeng_raw"].value
 
+        # NOTE: Update `batch_indices` to be end on a multiple of
+        # `HEAPS_PER_FENGINE_PER_CHUNK`, but only for the beam_outputs because
+        # they currently send `HEAPS_PER_FENGINE_PER_CHUNK` heaps all the time.
+        # This does not mean the final heap (for each beam_output) has sane
+        # data in it. In fact, ensure you verify data up to `n_beam_heaps - 1`.
```

Comment:
Why `n_beam_heaps - 1`? Shouldn't one just verify the values in `batch_indices`?

Reply:
Argh, too right. Both here and in your last comment, I'd made a mental `TODO` to update this explanation for something less 'hardcoded'.

Apologies, will update now-now.